### PR TITLE
State: Modularize Jetpack Connect state

### PR DESCRIPTION
**Pull request by tyxla**
This PR is one of the PRs that aim to modularize the state in Calypso. It is responsible for modularizing Jetpack Connect, a piece that is generally pretty independent from the rest of Calypso.

While it doesn't shave off too much from the main entry point, it's still a good improvement, because only the code that actually uses that state tree will load it, and this moves us a step forward towards a more modular Calypso. 

While the PR looks pretty big, it's very straightforward - it moves all actions and selectors into their dedicated files, moves out the redux tree to be independently initialized only by these actions/selectors, and moves the action type out of the main action types file. I'm happy to split to multiple PRs, but I believe that the atomic commits should make reviewing very straightforward.

For more details on state modularization, see #39261 and p4TIVU-9lM-p2.

#### Changes proposed in this Pull Request

* Add Jetpack Connect reducer initialization
* Initialize Jetpack Connect state in actions
* Initialize Jetpack Connect state in selectors
* Remove Jetpack Connect state from monolithic reducer
* Modularize Jetpack Connect action types
* Modularize Jetpack Connect actions
* Modularize Jetpack Connect selectors

#### Testing instructions

* Checkout this branch.
* Go to http://calypso.localhost:3000/read
* Verify that `state.jetpackConnect` is not defined.
* Go to http://calypso.localhost:3000/jetpack/connect
* Verify that * Verify that `state.jetpackConnect` is defined and you can see the redux state tree.
* Connect a Jetpack site starting from `/jetpack/connect` and verify it works well and produces no errors.
* Connect a Jetpack site starting from wp-admin and verify it works well and produces no errors in Calypso.

### CODE REVIEW

File: client/state/jetpack-connect/selectors/is-remote-site-on-sites-list.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Internal dependencies
+ */
+import { getAuthorizationData, getJetpackSiteByUrl } from 'state/jetpack-connect/selectors';
```
Comment (sgomes) [inline client/state/jetpack-connect/selectors/is-remote-site-on-sites-list.js]: More potential circular references here.

**Review by sgomes — COMMENTED** [review]
Thank you for taking on a good chunk of modularisation, @tyxla! The PR generally looks good, but I left some comments that could improve things.

File: client/state/jetpack-connect/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export { authorize } from './authorize';
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/state/jetpack-connect/actions/index.js]: Right back at me 😅 Thanks, fixed in https://github.com/Automattic/wp-calypso/pull/39759/commits/aae6db9f4891d830259ea4840f26ba856319ac6a

File: client/state/jetpack-connect/selectors/index.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export { getAuthAttempts } from './get-auth-attempts';
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/state/jetpack-connect/selectors/index.js]: Good call, fixed in https://github.com/Automattic/wp-calypso/pull/39759/commits/a0199438807d65536d5425b16402e5f20e483882

File: client/state/jetpack-connect/selectors/get-user-already-connected.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getAuthorizationData } from 'state/jetpack-connect/selectors';
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/state/jetpack-connect/selectors/get-user-already-connected.js]: Thanks, fixed all of them in https://github.com/Automattic/wp-calypso/pull/39759/commits/00ce9e5dacf055df71c1d110f545491bca5839e3

File: client/state/jetpack-connect/selectors/is-remote-site-on-sites-list.js (hunk 1 above)
Comment (tyxla) [reply client/state/jetpack-connect/selectors/is-remote-site-on-sites-list.js]: Thanks, fixed these too in in https://github.com/Automattic/wp-calypso/pull/39759/commits/00ce9e5dacf055df71c1d110f545491bca5839e3

**Comment by tyxla** [comment]
@sgomes thanks for your review 🙌 I've addressed the feedback, could you have another look when you get a chance? Thanks!

**Review by sgomes — APPROVED** [review]
LGTM, great work @tyxla! 👍

**Pull request merged**